test/test_linalg.py

```diff
@@ -1347,6 +1347,90 @@ def check(equation, operands, regex, exception=RuntimeError):
         check('a, ba', [x, y], r'operands do not broadcast with remapped shapes \[original->remapped\]: '
                                r'\[2\]->\[1, 2\] \[2, 3\]->\[2, 3\]')
 
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_qr_reduced(self, device, dtype):
+        t = torch.randn((7, 5), device=device, dtype=dtype)
```

Comment:
Test with a batch, too. This test can iterate over an iterable of shapes.

Reply:
I merged most of tests as you suggested in commit bbe2638289.
However, I had to write a separate version to test the batch because numpy doesn't support it: see e2231e9af8

Reply:
Awesome!